lib/uri_checker.rb

```diff
@@ -0,0 +1,213 @@
+require "uri"
+
+module UriChecker
+  class Report
```

Comment:
might want to add an `add_error` and `add_warning` method to this to save manipulating the attributes each time it's used